### PR TITLE
Add Cloudbuild step to run performance test by using the scenario test framework. 

### PR DESCRIPTION
**Pull request by gongmax**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:
Test cluster has not been created yet, it's blocked by the limit increase approval.

### CODE REVIEW

File: build/perf-test-image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+FROM gcr.io/cloud-builders/gcloud
```
Comment (markmandel) [inline build/perf-test-image/Dockerfile:1]: Doesn't the build image already have all these dependencies? Just questioning if we need another image to the set?

File: ci/perf-test-cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
```
Comment (markmandel) [inline ci/perf-test-cloudbuild.yaml:1]: Apache licence please.

File: build/perf-test-image/Dockerfile (hunk 1 above)
Comment (gongmax) [reply build/perf-test-image/Dockerfile:1]: Yes build image has the necessary dependencies but it also have many dependencies which is not required for this task. These dependencies add around 5 more minutes to the total build time. That's the only downside I see now. I wrote theses steps in this way to aline with what we did for e2e test, and I don't have strong objection to switch to build image though.

Comment (markmandel) [reply build/perf-test-image/Dockerfile:1]: You don't need to build the image though - you could pull the prebuilt image we build and cache, and run that instead.

Copy this section to do it:

https://github.com/googleforgames/agones/blob/93a4c4ab663d9a19f65959920d0a71408cabe457/cloudbuild.yaml#L76-L83

File: ci/perf-test-cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    id: pull-build-image
+    dir: build
+    env: ['REGISTRY=${_REGISTRY}', 'CUSTOM_LOCAL_TAG=performance-test']
+    args: [pull-build-image-with-tag]
+
+  #
+  # Run the performance tests with default feature gates setting
+  #
+  - name: performance-test
+    script: |
```
Comment (zmerlynn) [inline ci/perf-test-cloudbuild.yaml:47]: This "feels" long enough that we should maybe bump it out of YAML and into its own thing. WDYT?

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**